endpoint/endpoint.go

```diff
@@ -26,3 +26,17 @@ func Chain(outer Middleware, others ...Middleware) Middleware {
 		return outer(next)
 	}
 }
+
+// Failer is an interface that should be implemented by response types that
+// hold error properties as to separate business errors from transport errors.
+// If the response type can hold business errors it is highly advised to
+// implement Failer.
+// Response encoders can check if responses are Failer, and if so if they've
+// failed encode them using a separate write path based on the error.
+// Endpoint middlewares can test if a response type failed and also act or
+// report upon it.
+//
+// The addsvc example shows Failer's intended usage.
+type Failer interface {
+	Failed() error
+}
```

Comment:
Something rings... _weird_ to me about the names Failer/Failed, but I've thought about it for awhile and can't come up with anything better. Can everyone reading this PR have a brainstorm for a minute or two? It may be this is the best option, but I'd like to explore the space a little bit, collectively...